### DISCUSSION TITLE
Automatic folding or unfolding operation by stopping the movement of the mouse in the "folding mark" area.

### DISCUSSION THREAD

**Discussion by aaa1386** [freeplane/freeplane#1633] [Ideas]
Introduction: There are three methods to select a node, which you can choose in the path "Preferences -> Behavior -> Selection Method".
One way is to select a node by briefly stopping the mouse in the node area.
Ideas:
It is very interesting to have these three modes for "Folding Mark" in the software. .
 
User can set manual selection for node and automatic selection for "folding mark".
It can also adjust in reverse. Or the one that puts both in automatic selection mode.

The second idea to make the "Auto Expand" mode more complete:
If the "Auto Expand" mode is activated, it will automatically set the "Folding Mark" from the clicking mode to a mode where the folding or unfolding operation will be performed if the mouse pauses on the "Folding Mark" for a short time.
  Whenever the user disables "Auto Expand", the settings will return to the previous state.
In addition to being a new development for the entire software, this idea is also very helpful for the "Auto Expand" mode. It is enough to imagine this idea, I think you will like this idea too.

**Comment by aaa1386** [comment]
In my opinion, the "auto expand" mode will be much more complete and convenient if it is with the idea that was raised.
It is also used for other than "Auto Expand" mode.

**Reply by dpolivaev** [reply]
I do not know if it is useful. If some other users support your ideas I am going to consider them.

**Comment by aaa1386** [comment]
Three modes for node selection which are very useful in "Preferences -> Behavior -> Selection Method". If it wasn't useful, it wouldn't make sense to have it in the software. Therefore, the proposed idea of evolution is these three choices. It makes the three modes more complete. Instead of the user clicking on the "Folding Mark", the clicking action is done with a short pause.
In addition, the "Auto Expand" option becomes more complete and convenient with this idea. Because many times when we want the node and sub-branches to be manipulated by "Auto Expand" (or no manipulation), we have to click on "Folding Mark". Is it not useful if "Folding Mark" is clicked automatically?

**Comment by Illegit** [comment]
Hi All, I think this discussion touches on a level of deep habit in our thinking process when using maps. And therefore every answer means a slight change in personal approaches.
My personal feeling is that the suggestions seem interesting, I infer the following 3 modes:

>> 1. Manual Selection for Node: This mode likely involves the user manually clicking or selecting the node in the software.
>> 2. Automatic Selection for Folding Mark: In this mode, the “Folding Mark” (likely a feature or option in the software) is automatically selected or activated, possibly based on certain conditions or user actions.
>> 3. Automatic Selection for Both Node and Folding Mark: This mode seems to involve automatic selection or activation of both the node and the “Folding Mark”.
Automatizing the folding just with a pause could be interesting, but still, I do not perceive the current (FP1.11.9 10) effect of the auto-expand feature. Before supporting one position I need to have an explanation of the use of it. Thanks.

**Reply by Illegit** [reply]
I understood that the auto-expand works well only if you go with the keyboard, it isn't so effective if using mouse & keys in combination (let's say  as per my approach to maps).
Am I correct?

**Reply by dpolivaev** [reply]
I think it is a misunderstanding. Auto-expand folds only node which it unfolded earlier itself. Therefore if you want that it folds all nodes you can fold all nodes in the hierarchy at the beginning, there is a dedicated action for it. Afterwards there is no difference between using a keyboard and a mouse for navigation.

**Reply by Illegit** [reply]
OK thankyou. Understood now.

**Comment by aaa1386** [comment]
There are three options for node selection under "Preferences -> Behavior -> Selection Method".
If the proposed idea is to be prepared, these three options for "Folding Mark" must be created separately in the software.
This makes it possible for the user to choose between the way the "Folding Mark" is opened and the node selection in mixed modes, sometimes activating both, sometimes one, sometimes neither.
